examples/oidc/deployment.yaml

```diff
@@ -0,0 +1,87 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kube-rbac-proxy
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kube-rbac-proxy
+  namespace: default
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: kube-rbac-proxy
+subjects:
+- kind: ServiceAccount
+  name: kube-rbac-proxy
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kube-rbac-proxy
+rules:
+- apiGroups: ["authentication.k8s.io"]
+  resources:
+  - tokenreviews
+  verbs: ["create"]
+- apiGroups: ["authorization.k8s.io"]
+  resources:
+  - subjectaccessreviews
+  verbs: ["create"]
+- apiGroups: ["", "apps", "extensions", "rbac.authorization.k8s.io"]
+  resources: ["*"]
+  verbs: ["*"]
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["get", "list"]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: kube-rbac-proxy
+  name: kube-rbac-proxy
+spec:
+  ports:
+  - name: https
+    port: 8444
+    targetPort: https
+  selector:
+    app: kube-rbac-proxy
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: kube-rbac-proxy
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kube-rbac-proxy
+    spec:
+      # {BEGIN} for minikube development only if OIDC provider is deployed in minikube itself ie. dex
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      # {END} 
+      serviceAccountName: kube-rbac-proxy
+      containers:
+      - name: kube-rbac-proxy
+        image: docker.io/piotrmsc/kube-rbac-proxy:0.3.2
```

Comment:
Note to ourselfes: we should change this `docker.io/piotrmsc` once merged.

Reply:
let's change this now already to quay.io/brancz/kube-rbac-proxy:v0.4.0